option.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 PolyCrypt GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package payment
+
+import (
+	"perun.network/go-perun/wire/net"
+)
+
+// ClientOption is an option that can be passed to `NewClient`
+// to change the default behaviour of a client.
+type ClientOption map[string]interface{}
```

Comment:
Isn't an option more like a pair of a key and value? Why is it a map here?

Another option for realizing options that I have seen in the wild: https://www.sohamkamani.com/golang/options-pattern/

Reply:
I later-on do `mergeOptions([]ClientOption)` and then use the `dialer()` and `listener()` getters.  
By having it as a map it is easy to merge multiple options.  
We do it the same in *go-perun*, so i reused the concept.

Reply:
I use the approach from your link now, seems to work and is shorter.